evm_loader/lib/src/commands/emulate.rs

```diff
@@ -193,60 +345,121 @@ async fn emulate_trx<T: Tracer>(
         };
 
         let (exit_status, steps_executed, tracer) = evm.execute(step_limit, &mut backend).await?;
+
         if exit_status == ExitStatus::StepLimit {
             error!("Step_limit={step_limit} exceeded");
             return Ok((
                 EmulateResponse::revert(&NeonError::TooManySteps, &backend),
                 None,
             ));
         }
+        (exit_status, steps_executed, tracer)
+    };
+
+    calculate_response(
+        steps_executed,
+        exit_status,
+        storage,
+        tracer,
+        &emulate_request.provide_account_info,
+    )
+    .await
+}
+
+async fn emulate_trx_multiple_steps<'rpc, T: Tracer>(
```

Comment:
optional suggestion: how about reusing `emulate_trx_single_step` inside the `emulate_trx_multiple_steps`, is it possible?
You likely can move the call of `calculate_response` from the `emulate_trx_single_step` into the L273 and L279 

If it's too clunky/difficult to do, drop this suggestion.

Reply:
If I try to do so than 
`if exit_status == ExitStatus::StepLimit` must be outside of emulate_trx_single_step, otherwise it will give incorrect result for multiple steps where on almost every instruction you will receive step_limit.
Also, creation of backend & evm must be outside too because of the logic of a reinitialization (there is 

So, what's have left? Only call of evm::execute. I don't think it's a good idea